node-api/backend/genesis.go

```diff
@@ -35,3 +35,31 @@ func (b Backend) GenesisValidatorsRoot(slot math.Slot) (common.Root, error) {
 	}
 	return st.GetGenesisValidatorsRoot()
 }
+
+// GenesisForkVersion returns the genesis fork version of the beacon chain.
+func (b Backend) GenesisForkVersion(genesisSlot math.Slot) (common.Version, error) {
+	st, _, err := b.stateFromSlot(genesisSlot)
+	if err != nil {
+		return common.Version{}, errors.Wrapf(err, "failed to get state from slot %d", genesisSlot)
+	}
+	fork, err := st.GetFork()
+	if err != nil {
+		return common.Version{}, errors.Wrapf(err, "failed to get fork from state")
+	}
+	return fork.CurrentVersion, nil
```

Comment:
note: this is not the convention. This whole function should just be `ActiveForkVersionAtSlot(slot math.Slot)` and inside it should just call the chain spec method. No need to access state. Then in genesis handler, call with slot == 0.

Reply:
and this function on `Backend`: `ActiveForkVersionAtSlot` should not be in genesis.go, but probably backend.go

Reply:
`ActiveForkVersionForTimestamp(timestamp math.U64) common.Version`